storage/database/dynamodb.go

```diff
@@ -485,10 +489,11 @@ func (batch *dynamoBatch) Put(key, val []byte) error {
 	})
 	batch.size += dataSize
 
-	if len(batch.batchItems) == dynamoBatchSize {
+	if len(batch.batchItems) >= dynamoBatchSize {
+		thisTimeItems := batch.batchItems[:25]
+		batch.batchItems = batch.batchItems[25:]
 		batch.wg.Add(1)
-		dynamoWriteCh <- &batchWriteWorkerInput{batch.tableName, batch.batchItems, batch.wg}
-		batch.Reset()
+		dynamoWriteCh <- &batchWriteWorkerInput{batch.tableName, thisTimeItems, batch.wg}
```

Comment:
To reuse `batch` again,  you should remove batch items passed to a `batchWriteWorker` and reduce `batch.batchSize`.

Reply:
How about always setting the size to 0.
The size is used for writing oversized batch in levelDB. DynamoDB handles the oversized data inside the `Put()` function.

Reply:
I think it's not a good idea. Then, among `batch` implementations only dynamoDB will have a different semantic of `batch.ValueSize`. Even though DynamoDB handles overstacked items in `Put()` function, `batch.ValueSize` can still be used for overstacked items. Think about when `IdealBatchSize` is smaller than 25.

Reply:
@aidan-kwon I don't quite understand why we shouldn't set the size to 0 once we pass the data to the write channel. Could you please elaborate?

Reply:
@ehnuje winnie didn't suggest setting the size to 0 "once", but "always". Actually, the size was set to 0 in the original code by calling `batch.Reset()`. With this change, however, it should reduce the batch size after calculating the size of 25 items.

Reply:
@aidan-kwon As this is `Put` operation, it always meets the condition when `len(batch.batchItems) == dynamoBatchSize`, doesn't it? If so, always setting the size to 0 seems appropriate. In that case, we don't even need to slice the `batchItems`. We can just assign a new and empty slice for `batchItems`. (calls Reset?)

Reply:
@ehnuje There was an issue before making this change. What you are saying is same as the original code. However, the original code cannot request `Put` apporpriately. After storing after 10000 items, some items are either lost or stores more. There must be a error in this code. I couldn't find out what is the cause, but I will share after I do. :)

Reply:
I found out this code isn't the problem, my test code was. So, I reverted this part.
I also added some code that a critical will occur if there is a problem with user input. The retry logic will not be solve the problem in this case.
PTAL :)